docs/sql-data-sources-hive-tables.md

```diff
@@ -156,6 +159,28 @@ The following options can be used to configure the version of Hive that is used
     </td>
     <td>1.4.0</td>
   </tr>
+  <tr>
+    <td><code>spark.sql.hive.metastore.jars.path</code></td>
+    <td><code>(empty)</code></td>
+    <td>
+      Comma-separated paths of the jars that used to instantiate the HiveMetastoreClient.
+      This configuration is useful only when <code>spark.sql.hive.metastore.jars</code> is set as <code>path</code>. 
+      <br/>
+      The paths can be any of the following format:
+      <ol>
+        <li><code>file://path/to/jar/foo.jar</code></li>
+        <li><code>hdfs://nameservice/path/to/jar/foo.jar</code></li>
+        <li><code>/path/to/jar/</code>(path without URI scheme follow conf <code>fs.defaultFS</code>'s URI schema)</li>
```

Comment:
I remember there were discussion about using S3 as path in the original PR but it is not supported yet, is that correct? there could be edge cases where ppl set `fs.defaultFS` to a S3 path.

Also is viewfs supported as well? it's common ppl set that for `fs.defaultFS` as well.

Reply:
+1 on this, I just glanced at #29881 and it seems any Hadoop FileSystem should be allowed here rather than just `file`/`hdfs`?

Reply:
Thanks, @sunchao and @xkrogen .

1. For `S3`, in fact, I'm the one who asked about `S3` clarification. I didn't see any comment about not supporting it. I guess it's supported but not described because it requires `hadoop-cloud` module.
2. For `views`, I believe so.

BTW, this document came from the existing document of that `conf` description. For now, let's keep this identical with that. We can update both of them together later.

Reply:
Yes, sounds good to me. 

I remember @AngersZhuuuu mentioned some stackoverflow exception when using S3 as path but I don't know whether that is resolved afterwards.

Reply:
> Yes, sounds good to me.
> 
> I remember @AngersZhuuuu mentioned some stackoverflow exception when using S3 as path but I don't know whether that is resolved afterwards.

Yea, a stackoverflow seems caused by classloader. Not resolved yet since it rely on thirdparty jars.